src/authentication/authentication-hapi-plugin.ts

```diff
@@ -96,7 +96,7 @@ class AuthenticationHapiPlugin extends HapiPlugin {
     try {
       const credentials = request.auth.credentials as AccessToken;
       // The email has been validated in validateUser at this point
-      const user = await this.gitlab.getUserByEmail(credentials[subEmailClaimKey]!);
+      const user = await this.gitlab.getUserByEmailOrUsername(credentials.email!);
```

Comment:
This is called after logging in, right? If that's the case, the credentials (access token) might not contain the email. Shouldn't the sub be used here?